RecoTracker/LSTCore/standalone/code/core/trkCore.cc

```diff
@@ -423,18 +425,70 @@ std::vector<int> matchedSimTrkIdxs(std::vector<unsigned int> hitidxs,
     auto result = std::max_element(counts.begin(), counts.end());
     int rawidx = std::distance(counts.begin(), result);
     int trkidx = unique_idxs[rawidx];
-    if (trkidx < 0)
-      continue;
-    if (counts[rawidx] > (((float)nhits_input) * 0.75))
+    if (trkidx < 0) continue;
+    // std::cout <<  " counts[rawidx]: " << counts[rawidx] <<  std::endl;
+    // std::cout <<  " nhits_input: " << nhits_input <<  std::endl;
+    if (verbose) {
+      float fr = ((float)counts[rawidx]) / ((float)nhits_input);
+      std::cout <<  " fr: " << fr <<  std::endl;
+    }
+    // std::cout <<  " matchfrac: " << matchfrac <<  std::endl;
+    if (counts[rawidx] > (((float)nhits_input) * matchfrac)) {
       matched_sim_trk_idxs.push_back(trkidx);
+      matched_sim_trk_idxs_frac.push_back( ((float)counts[rawidx]) / ((float)nhits_input) );
+    }
     maxHitMatchCount = std::max(maxHitMatchCount, *std::max_element(counts.begin(), counts.end()));
   }
-  set<int> s;
+  map<int, float> pairs;
   unsigned size = matched_sim_trk_idxs.size();
-  for (unsigned i = 0; i < size; ++i)
-    s.insert(matched_sim_trk_idxs[i]);
-  matched_sim_trk_idxs.assign(s.begin(), s.end());
-  return matched_sim_trk_idxs;
+  for (unsigned i = 0; i < size; ++i) {
+    int idx = matched_sim_trk_idxs[i];
+    float frac = matched_sim_trk_idxs_frac[i];
+    if (pairs.find(idx) != pairs.end()) {
+      if (pairs[idx] < frac)
+          pairs[idx] = frac;
+    }
+    else {
+      pairs[idx] = frac;
+    }
+    // if (std::find(result.begin(), result.end(), idx) == result.end())
+    // {
+    //     result.push_back(idx);
+    //     result_frac.push_back(frac);
+    // }
+  }
+
+  vector<int> result;
+  vector<float> result_frac;
+
+  // Loop over the map using range-based for loop
+  for (const auto& pair : pairs) {
+    result.push_back(pair.first);
+    result_frac.push_back(pair.second);
+  }
+
+  // Sort indices based on 'values'
+  // auto indices = sort_indices(result_frac);
+
+  // // Reorder 'vec1' and 'vec2' based on the sorted indices
```

Comment:
still, why are those commented lines needed? If those are important, please uncomment them. Otherwise please drop them

Reply:
These lines sort the sim tracks by match fraction. I think that's not needed for our use case but correct me if I'm wrong